keyboards/preonic/keymaps/BTJ01/keymap-test.c

```diff
@@ -0,0 +1,250 @@
+ *                                    | -----+--^^--+------+------+--^^--+------+------+--^^--+------+------+------+------|
+ *                                    | Shift|  C←  |  C↓  |  C↑  |  C→  | Ins  |   (  |   )  |   1  |   2  |   3  |Shift |
+ *                                    |------+------+------+------+------+------+------+------+------+------+------+------|
+ *                                    | Ctrl | GUI  |Ct-Alt| Alt  |   FnSpace   |___FnSpace___|  .   |  0   |  =   |  _   |
+ *                                    '-----------------------------------------------------------------------------------'
+ */
+
+[RED] = LAYOUT_preonic_2x2u( \
+  KC_F1,    KC_F2,      KC_F3,      KC_F4,    KC_F5,        KC_F6,       KC_F7,      KC_F8,       KC_F9,   KC_F10,  KC_F11,  KC_F12,  \
+  _______,  KC_HOME,    KC_PGDN,    KC_PGUP,  KC_END,       KC_BSPC,     KC_MINS,    KC_PLUS,     KC_7,    KC_8,    KC_9,    KC_GRAVE,\
+  KC_ENT,   KC_LEFT,    KC_DOWN,    KC_UP,    KC_RGHT,      KC_DEL,      KC_SLSH,    KC_ASTR,     KC_4,    KC_5,    KC_6,    KC_QUOT, \
+  _______,  C(KC_LEFT), C(KC_DOWN), C(KC_UP), C(KC_RGHT),   KC_INS,      KC_LPRN,    KC_RPRN,     KC_1,    KC_2,    KC_3,    _______, \
+  _______,  _______,    _______,    _______,            _______,                _______,          KC_DOT,  KC_0,    KC_EQL,  KC_UNDS  \
+),                                     /*   |--------Fn-SPACE---------|_________Fn-SPACE_________|     */
+
+// 4 -- FN-CAPSLOCK
+
+[FC] = LAYOUT_preonic_2x2u( \
+  C(S(KC_ESC)), RESET,     DEBUG,    KC_NO,      KC_NO,        KC_NO,           KC_BRK,        KC_NLCK,     KC_SLCK, KC_PSCR,  KC_NO,  KC_NO,   \
+  KC_NO,        KC_NO,     KC_NO,    KC_NO,      KC_NO,        KC_NO,           KC_NO,         KC_NO,       KC_NO,   KC_NO,    KC_NO,  KC_NO,   \
+  _______,      KC_NO,     KC_NO,    KC_NO,      KC_NO,        KC_NO,           KC_NO,         KC_NO,       KC_NO,   KC_NO,    KC_NO,  EEP_RST, \
+  KC_NO,        KC_NO,     KC_NO,    KC_NO,      KC_NO,        KC_NO,           KC_NO,         KC_NO,       KC_NO,   KC_NO,    KC_NO,  KC_NO,   \
+  DF(QWRT),     DF(COLE),  KC_NO,    KC_NO,            _______,                        _______,             KC_NO,   KC_NO,    KC_NO,  KC_NO    \
+)                                          /*   | --------Fn-SPACE--------  | --------Fn-SPACE--------  |     */
+
+};
+
+#ifdef  AUDIO_ENABLE
+float start_up[][2] = {
+   MUSICAL_NOTE(_B5, 20),
+   MUSICAL_NOTE(_B6, 8),
+   MUSICAL_NOTE(_DS6, 20),
+   MUSICAL_NOTE(_B6, 8),
+};
+
+float tone_qwerty[][2]     = SONG(QWERTY_SOUND);
+float tone_dvorak[][2]     = SONG(DVORAK_SOUND);
+float tone_colemak[][2]    = SONG(COLEMAK_SOUND);
+
+float goodbye[][2] = SONG(GOODBYE_SOUND);
+#endif
+
+void persistent_default_layer_set(uint16_t default_layer) {
+  eeconfig_update_default_layer(default_layer);
+  default_layer_set(default_layer);
+}
```

Comment:
Could you remove this function?  

It's been deprecated in favor of the `set_single_persistent_default_layer` function (which is a core function).

https://docs.qmk.fm/#/ref_functions?id=setting-the-persistent-default-layer